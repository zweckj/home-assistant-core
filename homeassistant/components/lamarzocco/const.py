"""Constants for the La Marzocco integration."""

from typing import Final

DOMAIN: Final = "lamarzocco"

POLLING_INTERVAL: Final = 30
UPDATE_DELAY: Final = 3

DEFAULT_CLIENT_ID: Final = "7_1xwei9rtkuckso44ks4o8s0c0oc4swowo00wgw0ogsok84kosg"
DEFAULT_CLIENT_SECRET: Final = "2mgjqpikbfuok8g4s44oo4gsw0ks44okk4kc4kkkko0c8soc8s"

DEFAULT_PORT_LOCAL: Final = 8081

MODEL_GS3_AV: Final = "GS3 AV"
MODEL_GS3_MP: Final = "GS3 MP"
MODEL_LM: Final = "Linea Mini"
MODEL_LMU: Final = "Micra"

BREW_ACTIVE: Final = "brew_active"

GLOBAL: Final = "global"
MON: Final = "mon"
TUE: Final = "tue"
WED: Final = "wed"
THU: Final = "thu"
FRI: Final = "fri"
SAT: Final = "sat"
SUN: Final = "sun"

DAYS: Final = [MON, TUE, WED, THU, FRI, SAT, SUN]

MACHINE_NAME: Final = "machine_name"
SERIAL_NUMBER: Final = "serial_number"
