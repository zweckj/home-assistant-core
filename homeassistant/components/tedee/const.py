"""Constants for the Tedee integration."""
from datetime import timedelta

DOMAIN = "tedee"
NAME = "Tedee"

SCAN_INTERVAL = timedelta(seconds=10)

CONF_LOCAL_ACCESS_TOKEN = "local_access_token"
CONF_USE_LOCAL_API = "use_local_api"
CONF_USE_CLOUD_API = "use_cloud_api"
CONF_CLOUD_BRIDGE_ID = "cloud_bridge_id"
