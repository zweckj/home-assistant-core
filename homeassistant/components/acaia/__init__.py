"""Initialize the Acaia component."""
from homeassistant.config_entries import ConfigEntry
from homeassistant.const import CONF_MAC, CONF_NAME, Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import config_validation as cv
from homeassistant.helpers.typing import ConfigType

from .acaiaclient import AcaiaClient
from .const import CONF_IS_NEW_STYLE_SCALE, DOMAIN
from .coordinator import AcaiaApiCoordinator

CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)

PLATFORMS = [Platform.BUTTON, Platform.SENSOR]


async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
    """Set up the Acaia component."""
    hass.data.setdefault(DOMAIN, {})
    return True


async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
    """Set up Acaia as config entry."""

    name = config_entry.data[CONF_NAME]
    mac = config_entry.data[CONF_MAC]
    is_new_style_scale = config_entry.data.get(CONF_IS_NEW_STYLE_SCALE, True)

    acaia_client = AcaiaClient(
        hass, mac=mac, name=name, is_new_style_scale=is_new_style_scale
    )

    hass.data[DOMAIN][config_entry.entry_id] = coordinator = AcaiaApiCoordinator(
        hass, acaia_client
    )

    await coordinator.async_config_entry_first_refresh()

    await hass.config_entries.async_forward_entry_setups(config_entry, PLATFORMS)

    return True


async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
    """Unload a config entry."""

    unload_ok = await hass.config_entries.async_unload_platforms(
        config_entry, PLATFORMS
    )

    if unload_ok:
        hass.data[DOMAIN].pop(config_entry.entry_id)

    return unload_ok
