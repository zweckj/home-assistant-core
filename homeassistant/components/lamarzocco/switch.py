"""Switch platform for La Marzocco espresso machines."""
from collections.abc import Callable, Coroutine
from dataclasses import dataclass
from typing import Any

from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
from homeassistant.config_entries import ConfigEntry
from homeassistant.const import EntityCategory
from homeassistant.core import HomeAssistant
from homeassistant.helpers.entity_platform import AddEntitiesCallback

from .const import (
    DOMAIN,
    FRI,
    MODEL_GS3_AV,
    MODEL_GS3_MP,
    MODEL_LM,
    MODEL_LMU,
    MON,
    SAT,
    SUN,
    THU,
    TUE,
    WED,
)
from .entity import LaMarzoccoEntity, LaMarzoccoEntityDescription
from .lm_client import LaMarzoccoClient

DOSE = "dose"
ATTR_MAP_MAIN_GS3_AV = [
    (DOSE, "k1"),
    (DOSE, "k2"),
    (DOSE, "k3"),
    (DOSE, "k4"),
    (DOSE, "k5"),
]

ON = "on"
OFF = "off"
AUTO = "auto"
TIME = "time"

ATTR_MAP_AUTO_ON_OFF = [
    (MON, AUTO),
    (MON, ON, TIME),
    (MON, OFF, TIME),
    (TUE, AUTO),
    (TUE, ON, TIME),
    (TUE, OFF, TIME),
    (WED, AUTO),
    (WED, ON, TIME),
    (WED, OFF, TIME),
    (THU, AUTO),
    (THU, ON, TIME),
    (THU, OFF, TIME),
    (FRI, AUTO),
    (FRI, ON, TIME),
    (FRI, OFF, TIME),
    (SAT, AUTO),
    (SAT, ON, TIME),
    (SAT, OFF, TIME),
    (SUN, AUTO),
    (SUN, ON, TIME),
    (SUN, OFF, TIME),
]

TON = "ton"
TOFF = "toff"
PREBREWING = "prebrewing"
PREINFUSION = "preinfusion"

ATTR_MAP_PREBREW_GS3_AV = [
    (PREBREWING, TON, "k1"),
    (PREBREWING, TON, "k2"),
    (PREBREWING, TON, "k3"),
    (PREBREWING, TON, "k4"),
    (PREBREWING, TOFF, "k1"),
    (PREBREWING, TOFF, "k2"),
    (PREBREWING, TOFF, "k3"),
    (PREBREWING, TOFF, "k4"),
]

ATTR_MAP_PREINFUSION_GS3_AV = [
    (PREINFUSION, "k1"),
    (PREINFUSION, "k2"),
    (PREINFUSION, "k3"),
    (PREINFUSION, "k4"),
]

ATTR_MAP_PREINFUSION_LM = [
    (PREINFUSION, "k1"),
]

ATTR_MAP_PREBREW_LM = [
    (PREBREWING, TON, "k1"),
    (PREBREWING, TOFF, "k1"),
]


@dataclass
class LaMarzoccoSwitchEntityDescriptionMixin:
    """Description of an La Marzocco Switch."""

    control_fn: Callable[[LaMarzoccoClient, bool], Coroutine[Any, Any, None]]
    is_on_fn: Callable[[LaMarzoccoClient], bool]


@dataclass
class LaMarzoccoSwitchEntityDescription(
    SwitchEntityDescription,
    LaMarzoccoEntityDescription,
    LaMarzoccoSwitchEntityDescriptionMixin,
):
    """Description of an La Marzocco Switch."""


ENTITIES: tuple[LaMarzoccoSwitchEntityDescription, ...] = (
    LaMarzoccoSwitchEntityDescription(
        key="main",
        translation_key="main",
        icon="mdi:power",
        control_fn=lambda client, state: client.set_power(state),
        is_on_fn=lambda client: client.current_status["power"],
        extra_attributes={
            MODEL_GS3_AV: ATTR_MAP_MAIN_GS3_AV,
            MODEL_GS3_MP: None,
            MODEL_LM: None,
            MODEL_LMU: None,
        },
    ),
    LaMarzoccoSwitchEntityDescription(
        key="auto_on_off",
        translation_key="auto_on_off",
        icon="mdi:alarm",
        control_fn=lambda client, state: client.set_auto_on_off_global(state),
        is_on_fn=lambda client: client.current_status["global_auto"] == "Enabled",
        entity_category=EntityCategory.CONFIG,
        extra_attributes={
            MODEL_GS3_AV: ATTR_MAP_AUTO_ON_OFF,
            MODEL_GS3_MP: ATTR_MAP_AUTO_ON_OFF,
            MODEL_LM: ATTR_MAP_AUTO_ON_OFF,
            MODEL_LMU: ATTR_MAP_AUTO_ON_OFF,
        },
    ),
    LaMarzoccoSwitchEntityDescription(
        key="prebrew",
        translation_key="prebrew",
        icon="mdi:water",
        control_fn=lambda client, state: client.set_prebrew(state),
        is_on_fn=lambda client: client.current_status["enable_prebrewing"],
        entity_category=EntityCategory.CONFIG,
        extra_attributes={
            MODEL_GS3_AV: ATTR_MAP_PREBREW_GS3_AV,
            MODEL_LM: ATTR_MAP_PREBREW_LM,
            MODEL_LMU: ATTR_MAP_PREBREW_LM,
        },
    ),
    LaMarzoccoSwitchEntityDescription(
        key="preinfusion",
        translation_key="preinfusion",
        icon="mdi:water",
        control_fn=lambda client, state: client.set_preinfusion(state),
        is_on_fn=lambda client: client.current_status["enable_preinfusion"],
        entity_category=EntityCategory.CONFIG,
        extra_attributes={
            MODEL_GS3_AV: ATTR_MAP_PREINFUSION_GS3_AV,
            MODEL_LM: ATTR_MAP_PREINFUSION_LM,
            MODEL_LMU: ATTR_MAP_PREINFUSION_LM,
        },
    ),
    LaMarzoccoSwitchEntityDescription(
        key="steam_boiler_enable",
        translation_key="steam_boiler_enable",
        icon="mdi:water-boiler",
        control_fn=lambda client, state: client.set_steam_boiler_enable(state),
        is_on_fn=lambda client: client.current_status["steam_boiler_enable"],
        entity_category=EntityCategory.CONFIG,
    ),
)


async def async_setup_entry(
    hass: HomeAssistant,
    config_entry: ConfigEntry,
    async_add_entities: AddEntitiesCallback,
) -> None:
    """Set up switch entities and services."""

    coordinator = hass.data[DOMAIN][config_entry.entry_id]
    async_add_entities(
        LaMarzoccoSwitchEntity(coordinator, config_entry, description)
        for description in ENTITIES
        if not description.extra_attributes
        or coordinator.lm.model_name in description.extra_attributes
    )


class LaMarzoccoSwitchEntity(LaMarzoccoEntity, SwitchEntity):
    """Switches representing espresso machine power, prebrew, and auto on/off."""

    entity_description: LaMarzoccoSwitchEntityDescription

    async def async_turn_on(self, **kwargs: Any) -> None:
        """Turn device on."""
        await self.entity_description.control_fn(self._lm_client, True)
        await self._update_ha_state()

    async def async_turn_off(self, **kwargs: Any) -> None:
        """Turn device off."""
        await self.entity_description.control_fn(self._lm_client, False)
        await self._update_ha_state()

    @property
    def is_on(self) -> bool:
        """Return true if device is on."""
        return self.entity_description.is_on_fn(self._lm_client)
