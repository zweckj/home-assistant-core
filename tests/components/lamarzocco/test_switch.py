"""Tests for La Marzocco switches."""

from unittest.mock import MagicMock

import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.switch import (
    DOMAIN as SWITCH_DOMAIN,
    SERVICE_TURN_OFF,
    SERVICE_TURN_ON,
)
from homeassistant.const import ATTR_ENTITY_ID
from homeassistant.core import HomeAssistant
from homeassistant.helpers import device_registry as dr, entity_registry as er

pytestmark = pytest.mark.usefixtures("init_integration")


@pytest.mark.parametrize(
    (
        "entity_name",
        "method_name",
        "on_call",
        "off_call",
    ),
    [
        (
            "",
            "set_power",
            (True, None),
            (False, None),
        ),
        (
            "_auto_on_off",
            "enable_schedule_globally",
            (True,),
            (False,),
        ),
        (
            "_steam_boiler",
            "set_steam",
            (True, None),
            (False, None),
        ),
    ],
)
async def test_switches(
    hass: HomeAssistant,
    mock_lamarzocco: MagicMock,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
    entity_name: str,
    method_name: str,
    on_call: tuple,
    off_call: tuple,
) -> None:
    """Test the La Marzocco switches."""
    serial_number = mock_lamarzocco.serial_number

    control_fn = getattr(mock_lamarzocco, method_name)

    state = hass.states.get(f"switch.{serial_number}{entity_name}")
    assert state
    assert state == snapshot

    entry = entity_registry.async_get(state.entity_id)
    assert entry
    assert entry == snapshot

    await hass.services.async_call(
        SWITCH_DOMAIN,
        SERVICE_TURN_OFF,
        {
            ATTR_ENTITY_ID: f"switch.{serial_number}{entity_name}",
        },
        blocking=True,
    )

    assert len(control_fn.mock_calls) == 1
    control_fn.assert_called_once_with(*off_call)

    await hass.services.async_call(
        SWITCH_DOMAIN,
        SERVICE_TURN_ON,
        {
            ATTR_ENTITY_ID: f"switch.{serial_number}{entity_name}",
        },
        blocking=True,
    )

    assert len(control_fn.mock_calls) == 2
    control_fn.assert_called_with(*on_call)


async def test_device(
    hass: HomeAssistant,
    mock_lamarzocco: MagicMock,
    device_registry: dr.DeviceRegistry,
    entity_registry: er.EntityRegistry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test the device for one switch."""

    state = hass.states.get(f"switch.{mock_lamarzocco.serial_number}")
    assert state

    entry = entity_registry.async_get(state.entity_id)
    assert entry
    assert entry.device_id

    device = device_registry.async_get(entry.device_id)
    assert device
    assert device == snapshot
