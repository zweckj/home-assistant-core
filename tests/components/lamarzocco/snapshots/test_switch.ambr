# serializer version: 1
# name: test_auto_on_off
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'GS01234 Auto on/off',
      'icon': 'mdi:alarm',
    }),
    'context': <ANY>,
    'entity_id': 'switch.gs01234_auto_on_off',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_auto_on_off.1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'switch.gs01234_auto_on_off',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:alarm',
    'original_name': 'Auto on/off',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'auto_on_off',
    'unique_id': 'GS01234_auto_on_off',
    'unit_of_measurement': None,
  })
# ---
# name: test_auto_on_off.2
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': None,
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'lamarzocco',
        'GS01234',
      ),
    }),
    'is_new': False,
    'manufacturer': 'La Marzocco',
    'model': 'GS3 AV',
    'name': 'GS01234',
    'name_by_user': None,
    'serial_number': 'GS01234',
    'suggested_area': None,
    'sw_version': '1.1',
    'via_device_id': None,
  })
# ---
# name: test_main
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'GS01234 Main',
      'icon': 'mdi:power',
    }),
    'context': <ANY>,
    'entity_id': 'switch.gs01234_main',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_main.1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.gs01234_main',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:power',
    'original_name': 'Main',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'main',
    'unique_id': 'GS01234_main',
    'unit_of_measurement': None,
  })
# ---
# name: test_main.2
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': None,
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'lamarzocco',
        'GS01234',
      ),
    }),
    'is_new': False,
    'manufacturer': 'La Marzocco',
    'model': 'GS3 AV',
    'name': 'GS01234',
    'name_by_user': None,
    'serial_number': 'GS01234',
    'suggested_area': None,
    'sw_version': '1.1',
    'via_device_id': None,
  })
# ---
# name: test_prebrew[GS3 AV]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'GS01234 Prebrew',
      'icon': 'mdi:water',
    }),
    'context': <ANY>,
    'entity_id': 'switch.gs01234_prebrew',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_prebrew[GS3 AV].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'switch.gs01234_prebrew',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:water',
    'original_name': 'Prebrew',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'prebrew',
    'unique_id': 'GS01234_prebrew',
    'unit_of_measurement': None,
  })
# ---
# name: test_prebrew[GS3 AV].2
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': None,
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'lamarzocco',
        'GS01234',
      ),
    }),
    'is_new': False,
    'manufacturer': 'La Marzocco',
    'model': 'GS3 AV',
    'name': 'GS01234',
    'name_by_user': None,
    'serial_number': 'GS01234',
    'suggested_area': None,
    'sw_version': '1.1',
    'via_device_id': None,
  })
# ---
# name: test_prebrew[Linea Mini]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'LM01234 Prebrew',
      'icon': 'mdi:water',
    }),
    'context': <ANY>,
    'entity_id': 'switch.lm01234_prebrew',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_prebrew[Linea Mini].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'switch.lm01234_prebrew',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:water',
    'original_name': 'Prebrew',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'prebrew',
    'unique_id': 'LM01234_prebrew',
    'unit_of_measurement': None,
  })
# ---
# name: test_prebrew[Linea Mini].2
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': None,
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'lamarzocco',
        'LM01234',
      ),
    }),
    'is_new': False,
    'manufacturer': 'La Marzocco',
    'model': 'Linea Mini',
    'name': 'LM01234',
    'name_by_user': None,
    'serial_number': 'LM01234',
    'suggested_area': None,
    'sw_version': '1.1',
    'via_device_id': None,
  })
# ---
# name: test_prebrew[Micra]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'MR01234 Prebrew',
      'icon': 'mdi:water',
    }),
    'context': <ANY>,
    'entity_id': 'switch.mr01234_prebrew',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_prebrew[Micra].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'switch.mr01234_prebrew',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:water',
    'original_name': 'Prebrew',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'prebrew',
    'unique_id': 'MR01234_prebrew',
    'unit_of_measurement': None,
  })
# ---
# name: test_prebrew[Micra].2
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': None,
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'lamarzocco',
        'MR01234',
      ),
    }),
    'is_new': False,
    'manufacturer': 'La Marzocco',
    'model': 'Linea Micra',
    'name': 'MR01234',
    'name_by_user': None,
    'serial_number': 'MR01234',
    'suggested_area': None,
    'sw_version': '1.1',
    'via_device_id': None,
  })
# ---
# name: test_preinfusion[GS3 AV]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'GS01234 Preinfusion',
      'icon': 'mdi:water',
    }),
    'context': <ANY>,
    'entity_id': 'switch.gs01234_preinfusion',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_preinfusion[GS3 AV].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'switch.gs01234_preinfusion',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:water',
    'original_name': 'Preinfusion',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'preinfusion',
    'unique_id': 'GS01234_preinfusion',
    'unit_of_measurement': None,
  })
# ---
# name: test_preinfusion[GS3 AV].2
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': None,
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'lamarzocco',
        'GS01234',
      ),
    }),
    'is_new': False,
    'manufacturer': 'La Marzocco',
    'model': 'GS3 AV',
    'name': 'GS01234',
    'name_by_user': None,
    'serial_number': 'GS01234',
    'suggested_area': None,
    'sw_version': '1.1',
    'via_device_id': None,
  })
# ---
# name: test_preinfusion[Linea Mini]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'LM01234 Preinfusion',
      'icon': 'mdi:water',
    }),
    'context': <ANY>,
    'entity_id': 'switch.lm01234_preinfusion',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_preinfusion[Linea Mini].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'switch.lm01234_preinfusion',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:water',
    'original_name': 'Preinfusion',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'preinfusion',
    'unique_id': 'LM01234_preinfusion',
    'unit_of_measurement': None,
  })
# ---
# name: test_preinfusion[Linea Mini].2
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': None,
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'lamarzocco',
        'LM01234',
      ),
    }),
    'is_new': False,
    'manufacturer': 'La Marzocco',
    'model': 'Linea Mini',
    'name': 'LM01234',
    'name_by_user': None,
    'serial_number': 'LM01234',
    'suggested_area': None,
    'sw_version': '1.1',
    'via_device_id': None,
  })
# ---
# name: test_preinfusion[Micra]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'MR01234 Preinfusion',
      'icon': 'mdi:water',
    }),
    'context': <ANY>,
    'entity_id': 'switch.mr01234_preinfusion',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_preinfusion[Micra].1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': <EntityCategory.CONFIG: 'config'>,
    'entity_id': 'switch.mr01234_preinfusion',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:water',
    'original_name': 'Preinfusion',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'preinfusion',
    'unique_id': 'MR01234_preinfusion',
    'unit_of_measurement': None,
  })
# ---
# name: test_preinfusion[Micra].2
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': None,
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'lamarzocco',
        'MR01234',
      ),
    }),
    'is_new': False,
    'manufacturer': 'La Marzocco',
    'model': 'Linea Micra',
    'name': 'MR01234',
    'name_by_user': None,
    'serial_number': 'MR01234',
    'suggested_area': None,
    'sw_version': '1.1',
    'via_device_id': None,
  })
# ---
# name: test_steam_boiler_enable
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'friendly_name': 'GS01234 Steam boiler',
      'icon': 'mdi:water-boiler',
    }),
    'context': <ANY>,
    'entity_id': 'switch.gs01234_steam_boiler',
    'last_changed': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_steam_boiler_enable.1
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'switch',
    'entity_category': None,
    'entity_id': 'switch.gs01234_steam_boiler',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'name': None,
    'options': dict({
    }),
    'original_device_class': None,
    'original_icon': 'mdi:water-boiler',
    'original_name': 'Steam boiler',
    'platform': 'lamarzocco',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'steam_boiler',
    'unique_id': 'GS01234_steam_boiler_enable',
    'unit_of_measurement': None,
  })
# ---
# name: test_steam_boiler_enable.2
  DeviceRegistryEntrySnapshot({
    'area_id': None,
    'config_entries': <ANY>,
    'configuration_url': None,
    'connections': set({
    }),
    'disabled_by': None,
    'entry_type': None,
    'hw_version': None,
    'id': <ANY>,
    'identifiers': set({
      tuple(
        'lamarzocco',
        'GS01234',
      ),
    }),
    'is_new': False,
    'manufacturer': 'La Marzocco',
    'model': 'GS3 AV',
    'name': 'GS01234',
    'name_by_user': None,
    'serial_number': 'GS01234',
    'suggested_area': None,
    'sw_version': '1.1',
    'via_device_id': None,
  })
# ---
